src/variant_type.rs

```diff
@@ -208,6 +213,36 @@ impl ToOwned for VariantTy {
     }
 }
 
+impl StaticType for VariantTy {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl StaticType for VariantType {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl SetValue for VariantTy {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        gobject_ffi::g_value_take_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as glib_ffi::gpointer)
+    }
+}
+
+impl SetValueOptional for VariantTy {
+    unsafe fn set_value_optional(value: &mut Value, this: Option<&Self>) {
+        gobject_ffi::g_value_take_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as glib_ffi::gpointer)
```

Comment:
Is NULL a valid VariantTy?

Reply:
Probably not. I guess NULL would be ```"*"```?

Reply:
Needs to be checked in the GLib code, in the implementation of the boxed<->variantty code. Will check tomorrow if nobody else is faster :)

Reply:
It goes into `g_variant_type_copy()`, which says that it must not be `NULL`. So no `SetValueOptional` for both and you can implement `FromValue` for both

Reply:
So needed: `StaticType`, `FromValueOptional`, `FromValue` and `SetValue` for both types